crates/relayer/src/chain/namada/tx.rs

```diff
@@ -67,17 +71,46 @@ impl NamadaChain {
             txs.push((tx, signing_data));
         }
         let (mut tx, signing_data) = tx::build_batch(txs).map_err(NamadaError::namada)?;
+        let signing_data = signing_data.first().expect("SigningData should exist");
+
+        // Estimate the fee with dry-run
+        if let Some((fee_token, gas_limit, fee_amount)) =
+            self.estimate_fee(tx.clone(), &tx_args, signing_data)?
+        {
+            // Set the estimated fee
+            tx_args = tx_args
+                .fee_token(fee_token)
+                .gas_limit(gas_limit.into())
+                .fee_amount(
+                    fee_amount
+                        .to_string()
+                        .parse()
+                        .expect("Fee should be parsable"),
+                );
+            let fee_amount = rt
+                .block_on(signing::validate_fee(&self.ctx, &tx_args))
+                .map_err(NamadaError::namada)?;
+            rt.block_on(prepare_tx(
+                &tx_args,
+                &mut tx,
+                fee_amount,
+                relayer_key.secret_key.to_public(),
+            ))
+            .map_err(NamadaError::namada)?;
```

Comment:
there is no default gas limit?

Reply:
Without setting it explicitly, the default gas limit is applied for now.